src/main/java/com/bobocode/hoverla/bring/BringApplication.java

```diff
@@ -9,18 +20,99 @@
  *
  * Usage:
  * {@code
- *     ApplicationContext = BringApplication.loadContext();
- *     String bean = ApplicationContext.getBean("bean_name", String.class);
+ *     ApplicationContext context = BringApplication.loadContext("com.bobocode.hoverla.bring");
+ *     String bean = context.getBean("bean_name", String.class);
  * }
  * </pre>
+ * <p>
+ * BringApplication provides possibility to define log level. Use {@link ApplicationContextBuilder} for this purpose.
+ * <p>
+ * Usage:
+ * {@code
+ *      ApplicationContext context = BringApplication.getContextBuilder()
+ *              .logLevel(Level.DEBUG)
+ *              .build();
+ *      String bean = context.getBean("BeanName");
+ * }
+ *
+ * It is necessary to provide packages to be scanned to define bean definition configs.
  */
 public class BringApplication {
+    private BringApplication() {
+    }
+
     /**
      * Initializes and returns {@link ApplicationContext}
+     * @param packagesToScan packages to scan
+     */
+    public static ApplicationContext loadContext(String... packagesToScan) {
+        return getContext(packagesToScan);
+    }
+
+    /**
+     * Initializes input parameters (list of {@link BeanScanner}, {@link BeanDefinitionValidator}, {@link BeanInitializer})
+     * for ApplicationContext creation and return instance of context
+     *
+     * @param packagesToScan packages for scanning to define bean configs
+     * @return instance of {@link ApplicationContextImpl}
+     */
+    private static ApplicationContext getContext(String... packagesToScan)  {
+        // TODO: use this variable for scanners
+        String[] validatedPackages = validatePackagesToScan(packagesToScan);
+        // TODO: add initialization of scanners;
+        List<BeanScanner> scanners = List.of();
+
+        return new ApplicationContextImpl(scanners, new BeanDefinitionValidator(), new BeanInitializer());
+    }
+
+    private static String[] validatePackagesToScan(String... packagesToScan) {
```

Comment:
I think you should rewrite this method so that it doesn't allow null or empty packages at all. So, no need to return `String[]`, it should simply accept `packagesToScan` and perform its validation

Reply:
@NazarSmith228 changed